helix-term/src/commands/lsp.rs

```diff
@@ -104,24 +106,27 @@ struct SymbolInformationItem {
 
 impl ui::menu::Item for SymbolInformationItem {
     /// Path to currently focussed document
-    type Data = Option<lsp::Url>;
-
-    fn format(&self, current_doc_path: &Self::Data) -> Row {
-        if current_doc_path.as_ref() == Some(&self.symbol.location.uri) {
-            self.symbol.name.as_str().into()
-        } else {
-            match self.symbol.location.uri.to_file_path() {
-                Ok(path) => {
-                    let get_relative_path = path::get_relative_path(path.as_path());
+    type Data = Option<helix_core::Uri>;
+
+    fn format(&self, current_doc_uri: &Self::Data) -> Row {
+        match helix_core::Uri::try_from(&self.symbol.location.uri) {
+            Ok(uri) if uri.as_path().is_some() => {
+                if current_doc_uri
+                    .as_ref()
+                    .is_some_and(|doc_uri| doc_uri == &uri)
+                {
+                    self.symbol.name.as_str().into()
+                } else {
+                    let relative_path = path::get_relative_path(uri.as_path().unwrap());
                     format!(
                         "{} ({})",
                         &self.symbol.name,
-                        get_relative_path.to_string_lossy()
+                        relative_path.to_string_lossy()
                     )
                     .into()
                 }
-                Err(_) => format!("{} ({})", &self.symbol.name, &self.symbol.location.uri).into(),
             }
+            _ => format!("{} ({})", &self.symbol.name, &self.symbol.location.uri).into(),
         }
     }
```

Comment:
an `is_some` `unwrap` combo is a bit of an anti pattern. I think in this case is also not quite right: we only care if there is a path in the get_relative_path case. This was actual a review comment in an old pr. Also I think if let makes this read better.

```suggestion
    fn format(&self, current_doc_uri: &Self::Data) -> Row {
        if let Ok(uri) = helix_core::Uri::try_from(&self.symbol.location.uri) {
        	if current_doc_uri
                    .as_ref()
                    .is_some_and(|doc_uri| doc_uri == &uri)
            {
                return self.symbol.name.as_str().into()
            }
            if let Some(path) = uri.as_path() {
                 let relative_path = path::get_relative_path(uri.as_path().unwrap());
                 return format!(
                    "{} ({})",
                    &self.symbol.name,
                    relative_path.to_string_lossy()
                 ).into();
            }
        }
        format!("{} ({})", &self.symbol.name, &self.symbol.location.uri).into()
        }
    }
```